### DISCUSSION TITLE
Support return value when using CliRunner

### DISCUSSION THREAD

**Discussion by jayqi** [fastapi/typer#559] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typer.testing import CliRunner

app = typer.Typer()

@app.command()
def main():
    print("hello from stdout")
    return "hello from return_value"


runner = CliRunner()
result = runner.invoke(app)

result.exit_code
#> 0

result.stdout
#> 'hello from stdout\n'

result.return_value

result.return_value is None
#> True
```

<sup>Created at 2023-02-22 17:13:45 EST by [reprexlite](https://github.com/jayqi/reprexlite) v1.0.0</sup>

### Description

Click v8.0.0 introduced the ability to capture the return value of a Click command when using Click's CliRunner to the `return_value` attribute of the `Result` object that `CliRunner.invoke` returns.

When using Typer's CliRunner, the return value seems to get lost and is not captured. 

References:

- [PR in Click that implemented this](https://github.com/pallets/click/pull/1312)
- [Click `Result` API Reference](https://click.palletsprojects.com/en/8.1.x/api/#click.testing.Result)

### Operating System

macOS

### Typer Version

Typer 0.7.0

### Python Version

Python 3.9.16

### Additional Context

Click 8.1.3

**Comment by GiorgioSgl** [comment]
What about this?

This would be very helpful!

I'm in version 0.9.0 but still any update about this.

@jayqi Have you find any shortcut to solve the issue?

**Comment by GiorgioSgl** [comment, marked as answer]
Just found how to fix this just run the `invoke` method with `standalone_mode=False`

```python
def test_app():
    result = runner.invoke(
        app, ["command"], standalone_mode=False
    )

    assert result.exit_code == 0
    print(result.return_value) # print the result and not get None

```

Idea by [this](https://stackoverflow.com/questions/73507532/why-is-the-return-value-for-clirunner-invoke-object-null)
